src/rendergraph/common/rendergraph/material/rgbmaterial.cpp

```diff
@@ -0,0 +1,39 @@
+#include "rgbmaterial.h"
+
+#include <QVector2D>
+
+#include "rendergraph/materialshader.h"
+#include "rendergraph/materialtype.h"
+#include "rendergraph/uniformset.h"
+
+using namespace rendergraph;
+
+RGBMaterial::RGBMaterial()
+        : Material(uniforms()) {
+}
+
+/* static */ const AttributeSet& RGBMaterial::attributes() {
+    static AttributeSet set = makeAttributeSet<QVector2D, QVector3D>({"position", "color"});
+    return set;
+}
+
+/* static */ const UniformSet& RGBMaterial::uniforms() {
+    static UniformSet set = makeUniformSet<QMatrix4x4>({"ubuf.matrix"});
+    return set;
+}
+
+MaterialType* RGBMaterial::type() const {
+    static MaterialType type;
+    return &type;
+}
+
+int RGBMaterial::compare(const Material* other) const {
+    Q_ASSERT(other && type() == other->type());
+    const auto* otherCasted = static_cast<const RGBMaterial*>(other);
+    return otherCasted == this ? 0 : 1;
```

Comment:
This comes from Qt scene graph.

See 

https://doc.qt.io/qt-6/qsgmaterial.html#compare

I guess the question is: shouldn't this return -1 when appropriate?

Reply:
Ah this is a compare used for sorting. So you need to return -1 or 1 to "to minimize state changes" 
(I don't have a clue ...)

Reply:
Honestly I don't have a clue either and would need to dive into the Qt code to see what optimisation they achieve through this. I am ignoring it in the opengl backend.

Anyway, I am now wondering why I didn't simply do the pointer comparison in the scenegraph backend. And that makes we wonder why Qt isn't doing a pointer comparison internally (I guess because there are situations where there are more ideal comparison methods?). 

Interestingly, this comes from the Qt example code:
```
int CustomMaterial::compare(const QSGMaterial *o) const
{
    Q_ASSERT(o && type() == o->type());
    const auto *other = static_cast<const CustomMaterial *>(o);
    return other == this ? 0 : 1; // ### TODO: compare state???
}
```

That TODO seems to indicate that who implemented this example didn't have a clue either :-D

Now, from another example:

```
    int compare(const QSGMaterial *m) const override
    {
        const NoisyMaterial *other = static_cast<const NoisyMaterial *>(m);

        if (int diff = int(state.color.rgb()) - int(other->state.color.rgb()))
            return diff;

        if (!state.texture || !other->state.texture)
            return state.texture ? 1 : -1;

        const qint64 diff = state.texture->comparisonKey() - other->state.texture->comparisonKey();
        return diff < 0 ? -1 : (diff > 0 ? 1 : 0);
    }
```

Note the cast to NoisyMaterial ! And indeed the Qt docs read
```
The this pointer and other is guaranteed to have the same [type](https://doc.qt.io/qt-6/qsgmaterial.html#type)()
```

Reply:
In https://github.com/mixxxdj/mixxx/pull/13599/commits/9d55ab3daed44b0aaefe569533c9655e2d9b23dd I have removed all concrete material compares, and implemented a generic compare in rendergraph/Material::compare . Based on the Qt examples I think this is correct. The Qt examples need to compare concrete data, but rendergraph uses a generic cache for the uniform values, so I can leverage that.

please review and resolve this thread